website/docs/developers/transactions.md

```diff
@@ -0,0 +1,185 @@
+---
+sidebar_position: 5
+title: Transactions
+description: Understanding transaction types in the Mina protocol implementation
+slug: /developers/transactions
+---
+
+# Transactions
+
+## Overview
+
+Transactions in Mina represent state changes to the ledger. The protocol
+supports both user-initiated transactions and protocol-generated transactions,
+each serving distinct purposes in the blockchain's operation.
+
+This document provides an entry point for developers to understand how
+transactions are structured and processed in the Rust codebase.
+
+## Transaction Types
+
+The top-level `Transaction` enum represents all possible transactions in the
+Mina protocol:
+
+<!-- CODE_REFERENCE: ledger/src/scan_state/transaction_logic/mod.rs#L1093-L1100 -->
+
+```rust reference title="ledger/src/scan_state/transaction_logic/mod.rs"
+https://github.com/o1-labs/mina-rust/blob/develop/ledger/src/scan_state/transaction_logic/mod.rs#L1093-L1100
```

Comment:
This should be a permalink. If/when this becomes outdated, it's easier to figure out the changes if it's a permalink.

I see that there are others links in this PR too, same does for them.